src/Camera/QGCCameraControl.cc

```diff
@@ -1950,6 +1951,21 @@ QGCCameraControl::_handleDefinitionFile(const QString &url)
 {
     //-- First check and see if we have it cached
     QFile xmlFile(_cacheFile);
+
+    QString ftpPrefix(QStringLiteral("%1://").arg(FTPManager::mavlinkFTPScheme));
+    if (url.startsWith(ftpPrefix, Qt::CaseInsensitive)) {
+        int ver = static_cast<int>(_info.cam_definition_version);
+        QString fileName = QString::asprintf("%s_%s_%03d.xml",
```

Comment:
hey @dayjaby we have a question, we noticed you are saving the file here and then reading the byte array before submitting to `dataReady` we are concerned we are saving the file twice to disk, once when ftp manager downloads, and twice when the caching detection happens, is this why you are using the same filename?

Reply:
If I remember correctly the file got only transmitted once. But let me test that again.

Why should the cache detection save the file? The FTP manager saves it once, and from that moment onwards QGC should reuse the cached file.

Reply:
in http "downloads" we are saving the camera definition directly as a byte array, whereas on ftp you are saving it directly as a file then converting it to byte array, im wondering if you can either save as byte array directly, or double check the cache logic that happens after you signal data ready won't create a duplicate file locally

Reply:
You are right. Didn't expect "loadCameraDefinitionFile" to **save** a file.

With the newest commit, after successful ftp download, it just pretend the xml file is cached to prevent a second file save.